internal/restore/pvc_action.go

```diff
@@ -103,6 +104,10 @@ func (p *PVCRestoreItemAction) Execute(input *velero.RestoreItemActionExecuteInp
 		return nil, errors.WithStack(err)
 	}
 	p.Log.Infof("Starting PVCRestoreItemAction for PVC %s/%s", pvc.Namespace, pvc.Name)
+	if boolptr.IsSetToFalse(input.Restore.Spec.RestorePVs) {
+		p.Log.Infof("Restore did not request for PVs to be restored %s/%s", input.Restore.Namespace, input.Restore.Name)
+		return &velero.RestoreItemActionExecuteOutput{SkipRestore: true}, nil
```

Comment:
Do we want to skip the entire PVC here?

How about we allow the PVC to be restored, but without the change in datasource? We can skip resource for the Volumesnapshot and VolumesnapshotContent resource, but we should let the PVC be provisioned. It will have an empty volume which should be the intended behavior?

Actually, the flag "input.Restore.Spec.RestorePVs" should mean "input.Restore.Spec.RestorePVsFrom Snapshots" as we want this flag to only control if the data from snapshots needs to be restored, not the Persistent Volumes themselves?

Suggestion:
`return &velero.RestoreItemActionExecuteOutput{
			UpdatedItem: input.Item,
		}, nil`

Reply:
I think that makes sense. 
I will make this change as you suggested.

Reply:
Ok, if this is the approach then I think you should not be skipping restore of volumesnapshot and volumesnapshotcontent as well. 

Consider a case where the backup is from a restored PVC(CSI). The PVC will have a volumesnapshot as the datasource. Now, if this approach is implemented, the PVC will still have the old datasource pointing to some volumesnapshot, but, we are skipping the restoration of volumesnapshots. In this case, the restore will succeed but the PVC would never come up. Consider the case where the restore is ahppening to an alternate namespace or to a namespace that does not already have the volumesnapshot resource in the namespace. 

So, In my opinion, if we take this approach, we should remove the code to skip restore for Volumesnapshots and VolumeSnapshotcontents as well.

What do you think?

Reply:
"The PVC will have a volumesnapshot as the datasource. "

Why will that happen?

We will pretty much provision a clean PVC right, without any datasource. 
Since we are just returning the input.item as UpdateItem.

Basically, making it a no op.

Reply:
This can happen if you are trying to backup  a PVC which has been restored from a CSI snapshot earlier.

Reply:
Ah I see.
What if we add some code to handle this scenario

If restore PV = false, in the input item, we'll remove the Source.

Does that help?